homeassistant/components/recorder/db_schema.py

```diff
@@ -415,13 +418,21 @@ def shared_attrs_bytes_from_event(
         exclude_attrs = (
             exclude_attrs_by_domain.get(domain, set()) | ALL_DOMAIN_EXCLUDE_ATTRS
         )
-        if dialect == SupportedDialect.POSTGRESQL:
-            return json_bytes_strip_null(
-                {k: v for k, v in state.attributes.items() if k not in exclude_attrs}
-            )
-        return json_bytes(
+        encoder = json_bytes_strip_null if dialect == PSQL_DIALECT else json_bytes
+        bytes_result = encoder(
             {k: v for k, v in state.attributes.items() if k not in exclude_attrs}
         )
+        if len(bytes_result) > MAX_STATE_ATTRS_BYTES:
+            _LOGGER.warning(
+                "State attributes for %s exceed maximum size of %s bytes. "
+                "This can cause database performance issues; Attributes "
+                "will not be stored: %s",
+                state.entity_id,
+                MAX_STATE_ATTRS_BYTES,
+                state.attributes,
```

Comment:
we should not print the attributes, they are very big 🙈

Reply:
That makes sense. The user isn't going to be able to do anything about it right away.  👍 

I was thinking my audience was the dev.

Reply:
b6596ca6fe5a52c6c5e737bc6e0aefa97a5ff73b